src/isolate.rs

```diff
@@ -75,6 +75,18 @@ pub type NearHeapLimitCallback = extern "C" fn(
   initial_heap_limit: usize,
 ) -> usize;
 
+#[repr(C)]
+pub struct ModifyCodeGenerationFromStringsResult<'s> {
+  pub codegen_allowed: bool,
+  pub modified_source: Option<Local<'s, String>>,
+}
+
+pub type ModifyCodeGenerationFromStringsCallback<'s> =
+  extern "C" fn(
+    ctx: Local<'s, Context>,
+    code: Local<'s, Value>,
+  ) -> ModifyCodeGenerationFromStringsResult<'s>;
+
```

Comment:
I wonder if this is the correct definition of the callback. When exposing the callback from Deno (which is the next step up from this PR) I get linker errors.

![image](https://user-images.githubusercontent.com/1683034/90038994-5d967c00-dcc6-11ea-8a69-1dbf7cc252f1.png)

Any hints on how to do this properly? v8 passes two Locals and in the struct expects a MaybeLocal back (which I assume is equivalent to `Option<Local<'_, T>>`.

Reply:
@NeoLegends if this errors pops up when you tried to build Deno you'll need to run `cargo clean` before `cargo build`

Reply:
Added a test that tries to pass a callback, and: it fails in rusty_v8, too (and not only when building Deno). `cargo clean` doesn't help there.